app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,62 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.order(:id)
+  end
+
+  def show
+    @single_task = Task.find(params[:id])
+  end
+
+  def new
+    # give them a shell and invite them to fill out the data.  Allows introspection into the object in the view!
+    @single_task = Task.new
+  end
+
+  def create
+    @single_task = Task.new(task_create_params[:task])
+
+    if @single_task.save
+      redirect_to root_path
+    else
+      render :new
+    end
+
+  end
+
+  def edit
+    @single_task = Task.find(params[:id])
+  end
+
+  # update a specific task (both for complete and generic edit)
+  def update
+    @single_task = Task.update(params[:id], task_edit_params[:task])
+
+    if @single_task.save
+      redirect_to root_path
+    else
+      render :edit
+    end
+
+  end
+
+  # delete a task
+  def destroy
+    @single_task = Task.delete(params[:id]) # need a private method?
```

Comment:
I've never used a private method to pass in params for the destroy method, so I think how you have it here works.